File: API/Helpers/LinkParams.cs
namespace API.Helpers
{
    public class LinkParams
    {
        private const int MaxPageSize = 50;
        public int PageNumber { get; set; } = 1;
        private int _pageSize = 10;
        public int PageSize
        {
            get => _pageSize;
            set => _pageSize = (value > MaxPageSize ? MaxPageSize : value);
        }

        public int MinExpiryDate { get; set; } = 0;
        public int MaxExpiryDate { get; set; } = 365;
        public string OrderBy { get; set; } = "newest";
    }
}

File: client/src/app/_models/linkParams.ts
export class LinkParams {
    maxExpiryDate = 8640;
    pageNumber = 1;
    pageSize = 9;
    orderBy = 'newest';
    all = false;
    
    constructor(maxExpiryDate?: number, pageNumber?: number, pageSize?: number, orderBy?: string, all?: boolean){}
}